api/coreservice.go

```diff
@@ -1646,3 +1656,67 @@ func (core *coreService) TraceTransaction(ctx context.Context, actHash string, c
 	retval, receipt, err := core.SimulateExecution(ctx, addr, sc)
 	return retval, receipt, traces, err
 }
+
+// TraceCall returns the trace result of call
+func (core *coreService) TraceCall(ctx context.Context,
+	callerAddr address.Address,
+	blkNumOrHash any,
+	contractAddress string,
+	nonce uint64,
+	amount *big.Int,
+	gasLimit uint64,
+	data []byte,
+	config *logger.Config) ([]byte, *action.Receipt, *logger.StructLogger, error) {
+	var (
+		blkHash hash.Hash256
+		err     error
+	)
+	switch blkNumOrHash.(type) {
+	case uint64:
+		blkHash, err = core.dao.GetBlockHash(blkNumOrHash.(uint64))
+	case string:
+		blkHash, err = hash.HexStringToHash256(blkNumOrHash.(string))
+	default:
+		err = errors.New("invalid block number or hash")
+	}
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	getblockHash := func(height uint64) (hash.Hash256, error) {
+		return blkHash, nil
+	}
+	if gasLimit == 0 {
+		gasLimit = core.bc.Genesis().BlockGasLimit
+	}
+	traces := logger.NewStructLogger(config)
+	ctx = protocol.WithVMConfigCtx(ctx, vm.Config{
+		Debug:     true,
+		Tracer:    traces,
+		NoBaseFee: true,
+	})
+	ctx = genesis.WithGenesisContext(ctx, core.bc.Genesis())
```

Comment:
move L1705 to here
```
ctx, err = core.bc.Context(ctx)
	if err != nil {
		return nil, nil, nil, err
	}
```
since `bc.Context()` will include genesis inside